### DISCUSSION TITLE
How to memorize current directory A, go to dir B, and then back to A

### DISCUSSION THREAD

**Discussion by fritzrehde** [sxyazi/yazi#680] [1. Q&A] [answered]
I wanted to know what the default way of doing this in `yazi` is: I am in directory `A`. I have a file that I know is in directory `B` (which is quite far away from `A` in the file tree). I need to go to dir `B`, select a file to be moved, then go back to `A`, and finalize moving the file to `A`. My problem is: how do I "memorize" `A` with some keybinding. In `lf`, all `lf` processes share file selections, so I could just open a new `lf` instance, navigate to `B`, select it, close that new `lf` instance, and I'm still in `A` in my original `lf` instance, so I can just hit `paste`. But with `yazi`, I believe that workflow is not supported. Is there some way for me to mark the current dir as something, go elsewhere, and then quickly jump back to the marked dir?
Thanks!

**Comment by sxyazi** [comment, marked as answer]
See https://github.com/dedukun/bookmarks.yazi

**Reply by fritzrehde** [reply]
Thanks, I'll try this out!

**Comment by sxyazi** [comment]
BTW why not use Yazi's multi-tab feature? It can fulfill the behavior you described in lf, just without needing to create an unnecessary process.

See https://github.com/sxyazi/yazi/discussions/429#discussioncomment-7752151

**Reply by fritzrehde** [reply]
I wouldn't call it an "unnecessary process". I already use tmux, so I have a keybinding for quickly creating a new tab/window, and deleting it again. I usually prefer sticking to tmux for all my terminal multiplexing needs, which is why I also don't use the multiplexing features of the terminal itself (if it has those). I liked the way lf did it because it didn't force me to memorize another keybinding inside my terminal manager, it just fit in nicely with my tmux setup. Though, I also have no clue what other features yazi's multi-tab feature supports, so maybe I really am just unnecessarily negligent here.

**Reply by sxyazi** [reply]
Hmm I don't think this is contradictory to your tmux workflow. You can achieve this workflow through Yazi's plugin system; it just requires keeping track of the tab states (saved to a temporary file) and then restoring them through commands like `cd` or `arrow`, much like what [bookmarks.yazi](https://github.com/dedukun/bookmarks.yazi) above does. - I've plans to create such a plugin, but just haven't found the time.

On the other hand, unsupported multi-tab file managers force users to use tmux or other terminal multiplexers/window managers, because that's all it can do.
